playbooks/setup_jenkins_slave.yml

```diff
@@ -1,7 +1,49 @@
 ---
 - hosts: job_nodes
   user: root
+  vars:
+    rpc_release_file_name: "{{ lookup('env', 'WORKSPACE') ~ 'rpc-openstack/rpcd/etc/openstack_deploy/user_rpco_variables_defaults.yml' }}"
   tasks:
+
+    # We first check whether the file exists, as the job may
+    # not include a checkout of rpc-openstack
+    - name: Check if release file name exists
+      stat:
+        path: "{{ rpc_release_file_name }}"
+      register: _release_file
+
+    # We then want to check the value of the rpc_release
+    # variable in the file. It may be a bit of jinja, or
+    # not exist, or be a real value. These edge cases are
+    # needed to protect against due to the many releases
+    # tested by the gate and the varied approaches across
+    # them all.
+    # If the value is of the form r<major>.<minor>.<patch>
+    # and optionally including a release candidate suffix
+    # then we have a valid version.
+    - name: Determine rpc_release value
+      set_fact:
+        rpc_release_value: |-
+          lookup('pipe', 'awk "/^rpc_release/ { print $2; }" ' ~ rpc_release_file_name) | replace('"', '')
```

Comment:
Yes - if you look in the artifacts-14.0 branch you'll see a static value, which is what we're changing the whole release process to use. We don't use the value unless it matches a version string - see the conditional in the apt_repo task.